### PR TITLE
Braze

### PR DESCRIPTION
**Pull request by danijel**
- Add initial skeleton for Braze API integration with sendEmail action

### CODE REVIEW

File: extensions/braze/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Braze changelog
+
+## May 2024
+
+- Initial action to send an email
```
Comment (nckhell) [inline extensions/braze/CHANGELOG.md:5]:
```suggestion
- Extension added with two actions: Send email and Send SMS
```

File: extensions/braze/actions/sendEmail/config/fields.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { type Field, FieldType } from '@awell-health/extensions-core'
+import { type ZodTypeAny, z } from 'zod'
+
+export const fields = {
```
Comment (nckhell) [inline extensions/braze/actions/sendEmail/config/fields.ts:4]: @danijel How/where is the recipient defined?

File: extensions/braze/actions/sendEmail/config/fields.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    label: 'From Email',
+    description:
+      'Valid email address in the format "Display Name <email@address.com>"',
+    type: FieldType.STRING,
+    required: true,
+  },
+  body: {
+    id: 'body',
+    label: 'Email body',
+    description: 'Email body content(HTML)',
```
Comment (nckhell) [inline extensions/braze/actions/sendEmail/config/fields.ts:23]:
```suggestion
    description: 'Email body content',
```

File: extensions/braze/actions/sendEmail/config/fields.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    required: true,
+  },
+  from: {
+    id: 'from',
+    label: 'From Email',
+    description:
+      'Valid email address in the format "Display Name <email@address.com>"',
+    type: FieldType.STRING,
+    required: true,
+  },
```
Comment (nckhell) [inline extensions/braze/actions/sendEmail/config/fields.ts:19]: I'd rather split this up into two fields
- From email
- From name

The extension code merge the two together: `${fromName} <${fromEmail}>`

File: extensions/braze/actions/sendSms/config/fields.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import { type Field, FieldType } from '@awell-health/extensions-core'
+import { type ZodTypeAny, z } from 'zod'
+
+export const fields = {
```
Comment (nckhell) [inline extensions/braze/actions/sendSms/config/fields.ts:4]: Similar Q: how's the recipient defined?

File: extensions/braze/actions/sendEmail/config/fields.ts (hunk 2 above)
Comment (danijel) [reply extensions/braze/actions/sendEmail/config/fields.ts:4]: TBH, I don't think you define recipient, I think you define audience conditions and that it sends messages to all the people who satisfy these conditions, so I think we'll need a json audience field, or something similar where we build json our selves. API example uses audience, here is full example https://www.braze.com/docs/api/endpoints/messaging/send_messages/post_send_messages/#example-request

File: extensions/braze/actions/sendSms/config/fields.ts (hunk 5 above)
Comment (danijel) [reply extensions/braze/actions/sendSms/config/fields.ts:4]: same answer as for email, I'll ask John once he gets in touch about api key

**Pull request closed without merging**